src/main/scala/query/Assignment.scala

```diff
@@ -0,0 +1,53 @@
+package com.originate.scalypher
+
+import com.originate.scalypher.path.Node
+import com.originate.scalypher.types.Referenceable
+import com.originate.scalypher.types.ReferenceableMap
+import com.originate.scalypher.where.Reference
+import com.originate.scalypher.where.ReferenceWithProperty
+import com.originate.scalypher.where.ObjectReference
+import com.originate.scalypher.where.ValueReference
+
+sealed trait Assignment extends ToQueryWithIdentifiers {
+  def referenceables: Set[Referenceable]
+}
+
+case class RemovePropertyAssignment(reference: ReferenceWithProperty) extends Assignment {
+  def referenceables: Set[Referenceable] =
+    reference.getReferenceable.toSet
+
+  def toQuery(referenceableMap: ReferenceableMap): String =
+    s"${reference.toQuery(referenceableMap)} = NULL"
+}
+
+trait MergeableProperties { self: Referenceable =>
```

Comment:
oic, "merge" means "merge this set of properties with what's already there"? merge as joining collections together is clear enough, but i didn't realize it's merging into something in the db, rather than another collection of properties. so i think a better name would express that in particular.